src/main/java/duke/tools/Parser.java

```diff
@@ -0,0 +1,142 @@
+
+    /**
+     * Private constructor to prevent object creation.
+     */
+    private Parser() {}
+
+    /**
+     * Returns the Command representing the user's input.
+     *
+     * @param str The input String from the user.
+     * @return The Command representing the user's input.
+     * @throws DukeException
+     */
+    public static Command parseCommand(String str) throws DukeException {
+        try {
+            String[] inputLine = str.split(" ", 2);
+            switch (inputLine[0]) {
+            case "bye":
+                return new ByeCommand();
+            case "list":
+                return new ListCommand();
+            case "mark":
+                return new MarkCommand(parseTaskIndex(inputLine[1].strip()));
+            case "unmark":
+                return new UnmarkCommand(parseTaskIndex(inputLine[1].strip()));
+            case "delete":
+                return new DeleteCommand(parseTaskIndex(inputLine[1].strip()));
+            case "find":
+                return new FindCommand(inputLine[1].strip());
+            case "todo":
+                return new TodoCommand(inputLine[1].strip());
+            case "deadline":
+                return parseDeadlineCommand(inputLine[1].strip());
+            case "event":
+                return parseEventCommand(inputLine[1].strip());
+            default:
+                throw new DukeException("Exception: Unknown command.");
+            }
+        } catch (NumberFormatException e) {
+            throw new DukeException("Exception: Incorrect command parameters.");
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Exception: Insufficient command parameters.");
+        }
+    }
+
+    /**
+     * Returns the index of the Task the user tries to access.
+     *
+     * @param str The input String from the user.
+     * @return The index of the Task the user tries to access.
+     */
+    private static int parseTaskIndex(String str) {
+        return Integer.parseInt(str) - 1;
+    }
+
+    /**
+     * Makes sense of the user's input to add a task with deadline.
+     *
+     * @param str The input String from the user specifying the parameters of the task with deadline.
+     * @return The DeadLineCommand representing the user's input
+     * @throws DukeException
+     */
+    private static DeadlineCommand parseDeadlineCommand(String str) throws DukeException {
+        try {
+            String[] descDateTime = str.split("/by", 2);
+            return new DeadlineCommand(descDateTime[0].strip(), parseDateTime(descDateTime[1].strip()));
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Exception: Insufficient command parameters.");
+        }
+    }
+
+    /**
+     * Makes sense of the user's input to add an event.
+     *
+     * @param str The input String from the user specifying the parameters of the event.
+     * @returnThe EventCommand representing the user's input
```

Comment:
Might there be an intention to put a space between the param and the description?

Reply:
Thanks for pointing out my styling errors in such detail! I would definitely have missed that myself. 😄